### DISCUSSION TITLE
REL Integration into multi-linker framework

### DISCUSSION THREAD

**Discussion by kmdn** [informagi/REL#118] [Q&A]
Hey there!

Love the linker!

I'd like to include (parts of) REL in an entity linking framework I've been working on for a while - if the devs are fine with it, of course. I'm working on a framework dividing up linkers into 3 main steps (additionally to a bunch of interactions through merging/splitting capabilities): mention detection, candidate generation and entity disambiguation.

At an initial glance, it seems a tad "difficult" to separate those steps in REL despite your wonderful Youtube video showing that it's basically "multiple awesome components" executed after one another. Could you perchance give me some pointers how to potentially achieve my task or add API functionality for calling on different sub-tasks?

Thanks again for the great work & wishing a great week!

Best regards,
kmdn

**Comment by arjenpdevries** [comment]
REL's architecture already uses the same three components you mention. It's already possible to use the API with and without a given mention. It shouldn't be too hard to also make a call for the last step only, this doesn't have a very high priority here though... What's the use case, easing research or do you have a "real" application in mind?

**Reply by kmdn** [reply]
(Sorry for the late reply!)

Thank you for your reply!

The use case here would be mainly for future research purposes - and to speed up research/development of novel approaches.

It doesn't seem like there is a "proper standard" in the field yet in regards to these substeps for EL pipelines (please correct me if I'm wrong).
As the subsequent steps in a pipeline are highly dependant on their preceding ones, it would stand to reason (assumption) that varying these, would potentially affect pipeline results. I'd like to research these effects on one hand and find if it makes sense to run all kinds of "fancy" combinations, e.g. allowing for more robust setups in specialised settings, assuming domain-dependence of certain systems.

Therefore, I am trying to have a range of pipelines and steps thereof 'easily executable' and combinable with pre-existing ones.

Would it be fine to use REL in such a framework to that end / would it make sense to you for it to be included?

Kind regards,
kmdn
